### ISSUE TITLE
"Show All (x)" Feature in Multi Input Control

### ISSUE THREAD

**Issue by D034394** [SAP/fundamental-ngx#2566]
Dear Fundamentals Team,

we are using the component "Multi Input" quite intensively.
https://sap.github.io/fundamental-ngx/#/core/multi-input

But we haven't figured out yet what the additional feature of "Show All (x)" is supposed to do. We do not see any documentation. Playing around with the provided examples on the page mentioned above ends nowhere.

We kindly ask to point us to proper documentation of that feature or to briefly explain here what it is doing.

Many thanks,
Kai
![image](https://user-images.githubusercontent.com/52921101/82838384-584a4200-9ecc-11ea-9827-93716e26948d.png)

**Comment by JKMarkowski** [comment]
Helo @D034394,

Show all button can be used, to clear search text from input, we added way to remove it by passing `[showAllButton]="false"`, but it's not released yet.

![multiuinput](https://user-images.githubusercontent.com/26483208/83020260-847ad580-a028-11ea-9fa4-74d48483a2ac.gif)

**Comment by D034394** [comment]
Hi there, many thanks for the information. I just figured out that your multi input control is missing an important feature compared to [corresponding UI5 control](https://sapui5.hana.ondemand.com/#/entity/sap.m.MultiComboBox/sample/sap.m.sample.MultiComboBox). WHen clicking on the "x more" button in case you already have multiple entries selected you get a drop-down list of all already selected items. In that case your "Show All" button makes perfectly sense and would be even an additional feature which UI5 does not have. So, please correct the "x more" behavior. Many thanks
![2020-05-28_12-19-06](https://user-images.githubusercontent.com/52921101/83130494-ac2c7500-a0de-11ea-8275-e794085000f4.gif)

**Comment by D034394** [comment]
Hi @JKMarkowski, @droshev do you have any feedback to to the suggested improvement I provided above? Are you guys working on that? Do you want me to create a new ticket for that? Anything else needed from my side?
Many thanks, Kai

**Comment by JKMarkowski** [comment]
Hello @D034394,

We had some discussion about the behaviour of multi input with tokenizer. We will provide the way to toggle `x more` as it is on ui5. I will let you know when it's ready to check.

**Comment by mikerodonnell89** [comment]
Hi @D034394 , I will be fixing the "____ more" cozy mode issue as a part of #2575 and it should be ready shortly.  Thank you for raising this issue

**Comment by D034394** [comment]
Hi @mikerodonnell89, do you have any updates?

**Comment by D034394** [comment]
Hi @mikerodonnell89, @droshev do you guys have any feedback on this issue? Will the issue I addressed [here](https://github.com/SAP/fundamental-ngx/issues/2566#issuecomment-635257057) will be fixed soon?

**Comment by mikerodonnell89** [comment]
Hi @D034394, sorry for the late response.  The "show all" button displays when the user has typed something in the input, filtering the list of selectable items.  Clicking "show all" clears the input text, resetting the filter and displaying every option.  This has been released

**Comment by D034394** [comment]
Hi Mike (@mikerodonnell89 ),
Thanks, but I know this already. This is what @JKMarkowski already explained in his comment on May 27th (see above) and what already worked before I created this issue. But this behavior is not compliant with the [corresponding UI5 control](https://sapui5.hana.ondemand.com/#/entity/sap.m.MultiComboBox/sample/sap.m.sample.MultiComboBox). I am waiting for the feature which I explained [here](https://github.com/SAP/fundamental-ngx/issues/2566#issuecomment-635257057). Clicking on "x More" would open the drop down with all selected options (non-selected are hidden). Clicking on "See All" would now open the complete list and I could select additional options. But clicking on "x More" and getting the complete list including the options I had not selected doesn't make sense.
Is anyone working on this? Or did I waste time waiting for it now?
Thanks, Kai

**Comment by droshev** [comment]
@D034394 Thank you for raising this issue. In fact there are few issues that need to be addressed. I hope you can have a little more patience. Your input is very valuable here.

**Comment by mikerodonnell89** [comment]
Hi @D034394 apologies as I did not understand your initial request.  Working on addressing these issues now

**Comment by mikerodonnell89** [comment]
Have fixed this as a part of https://github.com/SAP/fundamental-ngx/pull/2870

**Comment by droshev** [comment]
@D034394 can you check out https://fundamental-ngx.netlify.app/

**Comment by D034394** [comment]
Hi Deno (@droshev), Mike (@mikerodonnell89),
many thanks for your messages. I just checked out the link with the updated component.
First of all I got a bit confused on the [Simple Multi Input](https://fundamental-ngx.netlify.app/#/core/multi-input#simpleMultiInput) by being able to see tokens smoothly sliding out of the field on the left side when there is not enough space (overflowing, e.g. manually reducing the width of the browser window). These overflowing tokens can still be reviewed by swiping the tokens left and right. But I guess I just have never seen this swiping before. So far, I was more used to the behavior shown in the [Compact Multi Input](https://fundamental-ngx.netlify.app/#/core/multi-input#compactMultiInput) control where tokens simply disappear and the "more" counter increases.
Generally, I like the implemented Show All (x) feature. But I think there are still a few bugs:
(1) There is some whitespace on the right of the dropdown which increases if the field grows in width.
(2) The "Show All" row hight is very small.
![image](https://user-images.githubusercontent.com/52921101/88409452-448c6080-cdd5-11ea-87c0-df38ae264bf3.png)

**Comment by droshev** [comment]
@mikerodonnell89 I can reproduce both issues @D034394 listed.
<img width="554" alt="Screen Shot 2020-07-24 at 9 19 47 PM" src="https://user-images.githubusercontent.com/4380815/88445848-73b1ca80-cdf3-11ea-9be6-962ab9b379fe.png">
<img width="715" alt="Screen Shot 2020-07-24 at 9 18 44 PM" src="https://user-images.githubusercontent.com/4380815/88445849-74e2f780-cdf3-11ea-8155-4d6a1e706225.png">

@D034394 more about the swiping can be found [here](@fundamental-styles/s/docs/component-card)

**Comment by mikerodonnell89** [comment]
Hi @D034394, the swiping only happens in cozy mode which is intended only for mobile and tablet use.  For desktop, use the [compact] flag.  These are the ux specifications coming from the designers.  Thank you for pointing out the visual issues, seems like these came a bit later.  I have addressed them in this pull request: https://github.com/SAP/fundamental-ngx/pull/2924

**Comment by D034394** [comment]
Hi Mike @mikerodonnell89 many thanks for your update and work.
When can we expect the fixes being released? We have implemented the Multi Input control already and get a lot of complaints.
Other question: we have decided long time ago that we generally use Cozy for all devices, so also desktop. So far, this was not an issue for us as it mainly has visual impacts which we intended. But with this control we see now some negative impact with swiping not useable on desktop and arrow key navigation from left to right and vice versa not possible.
Is there any chance that we can make the cozy control recognize that it is used on a desktop and then behaves like the compact version of it?
cc deno @droshev
